src/engine/strat_engine/metadata.rs

```diff
@@ -184,13 +184,14 @@ impl StaticHeader {
 
 
     /// Read data from f into buffer of length _BDA_STATIC_HDR_SIZE.
-    fn read_into_buf<F>(f: &mut F) -> EngineResult<[u8; _BDA_STATIC_HDR_SIZE]>
+    /// Return a newly constructed buffer and the number of bytes read.
+    fn read_into_buf<F>(f: &mut F) -> EngineResult<([u8; _BDA_STATIC_HDR_SIZE], usize)>
         where F: Read + Seek
     {
         try!(f.seek(SeekFrom::Start(0)));
         let mut buf = [0u8; _BDA_STATIC_HDR_SIZE];
-        try!(f.read(&mut buf));
-        Ok(buf)
+        let bytes = try!(f.read(&mut buf));
```

Comment:
That's what clippy suggested, so I tried it, then, when the existing tests failed I realized what was wrong. (I guess it is an archaism that the branch is called ```master-read-exact``` still). 

What is wrong is that ```setup()``` is called in a context where it is not certain whether there is a static header or not. One perfectly good reason for there not being a static header is if we have some peculiar block device which we can't read a lot of data from and which does not belong to Stratis. The result of using ```read_exact()``` is to cause engine initialization to promptly fail, presumably in ```find_all``` when it tries to determine ownership on some slightly surprising device.